cql3/statements/create_keyspace_statement.cc

```diff
@@ -119,6 +121,21 @@ future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector
                 warnings.push_back("Keyspace `initial` tablets option is deprecated.  Use per-table tablet options instead.");
             }
         }
+
+        // If `rf_rack_valid_keyspaces` is enabled, it's forbidden to create an RF-rack-invalid keyspace.
+        // Verify that it's RF-rack-valid.
+        // For more context, see: scylladb/scylladb#23071.
+        if (cfg.rf_rack_valid_keyspaces()) {
+            try {
+                // We hold a group0_guard here, so it's correct to check this here.
+                // There's no ongoing schema or topology change.
```

Comment:
I don't think it's true that there's no ongoing topology change. Topology changes are multi-step sagas which only hold group0 guard briefly around certain transition points.